### PR TITLE
feat(Utility): provide remaining and elapsed time when requested

### PR DESCRIPTION
**Pull request by fight4dream**
The CountdownTimer now has RequestElapsedTime and RequestRemainingTime methods
which emit the ElapsedTime and RemainingTime of the timer.

### CODE REVIEW

**Comment by thestonefox** [comment]
Requires tests adding to `CountdownTimerTest`

**Comment by fight4dream** [comment]
Tests provided

**Comment by thestonefox** [comment]
The scope of this should be `Utility` as it's in that namespace

so `feat(Utility): ...`

Also the `TimerEmitTime` test is not consistent, it will sometimes fail with:

```
TimerEmitTime (0.524s)
---
UnityEngine.Assertions.AssertionException : Assertion failure. Values are not equal.
Expected: 0.5 == 0.4999999
---
at UnityEngine.Assertions.Assert.Fail (System.String message, System.String userMessage) [0x0003c] in C:\buildslave\unity\build\Runtime\Export\Assertions\Assert\AssertBase.cs:29 
  at UnityEngine.Assertions.Assert.AreEqual[T] (T expected, T actual, System.String message, System.Collections.Generic.IEqualityComparer`1[T] comparer) [0x00061] in C:\buildslave\unity\build\Runtime\Export\Assertions\Assert\AssertGeneric.cs:31 
  at UnityEngine.Assertions.Assert.AreEqual[T] (T expected, T actual, System.String message) [0x00009] in C:\buildslave\unity\build\Runtime\Export\Assertions\Assert\AssertGeneric.cs:19 
  at UnityEngine.Assertions.Assert.AreEqual[T] (T expected, T actual) [0x00004] in C:\buildslave\unity\build\Runtime\Export\Assertions\Assert\AssertGeneric.cs:13 
  at Test.Zinnia.Utility.CountdownTimerTest+<TimerEmitTime>d__10.MoveNext () [0x00339] in D:\Users\habal\Documents\Unity Projects\ExtendReality\Zinnia.Unity\Assets\Zinnia.Unity\Tests\Editor\Utility\CountdownTimerTest.cs:383 
  at UnityEngine.TestTools.TestEnumerator+<Execute>d__5.MoveNext () [0x0004c] in D:\Users\habal\Documents\Unity Projects\ExtendReality\Zinnia.Unity\Library\PackageCache\com.unity.test-framework@1.0.13\UnityEngine.TestRunner\NUnitExtensions\Attributes\TestEnumerator.cs:31
```

File: Runtime/Utility/CountdownTimer.cs [hunk 1]
```diff
@@ -53,13 +73,59 @@ public bool IsRunning
...
+        /// <summary>
+        /// <see cref="Time.time"/> when <see cref="Begin"/> is called.
+        /// </summary>
+        protected float beginTime;
+        /// <summary>
+        /// <see cref="Time.time"/> of the current frame.
+        /// </summary>
+        protected float currentTime;
+
+        protected virtual void SetInternalStates()
```
Comment (thestonefox) [inline Runtime/Utility/CountdownTimer.cs]: requires a comment as to what the method does, also protected methods appear after public methods.

So move this before the `Complete` on line 194 then it should be good to go.

**Pull request merged**